tools/bigquery-zos-mainframe-connector/gszutil/src/main/scala/com/google/cloud/bqsh/BQ.scala

```diff
@@ -38,10 +38,15 @@ object BQ extends Logging with GoogleApiL2Retrier {
   override val retriesTimeoutMillis: Int = Services.l2RetriesTimeoutMillis
 
   def genJobId(projectId: String, location: String, zos: MVS, jobType: String, suffix: String): JobId = {
```

Comment:
There is a [method in this class](https://github.com/jasonmar/professional-services/blob/mainframe-connector-a/tools/bigquery-zos-mainframe-connector/gszutil/src/main/scala/com/google/cloud/bqsh/BQ.scala#L57) with the same name and slightly different signature. Should't we also update that one to disallowed character in job name or step name by filtering during job id creation?

Reply:
consolidated into a single method with optional random suffix flag